### DISCUSSION TITLE
How to use the InvalidateCrosshair method in CoreAxis.cs?

### DISCUSSION THREAD

**Discussion by Eekkow** [Live-Charts/LiveCharts2#1636] [Q&A]
I'm currently working with a charting library and came across the InvalidateCrosshair method in the CoreAxis class. The method signature is as follows:

![Screenshot 2024-10-07 at 10 41 20](https://github.com/user-attachments/assets/2d8ddeaf-3bc6-42a3-89f2-c34f7aaf89ee)

I understand that this method is used to update the crosshair on a chart, but I'm not sure how to use it correctly. Could someone provide an example of how to call this method and explain what the chart and pointerPosition parameters represent? Any help would be appreciated.

**Comment by beto-rodriguez** [comment]
Hello

This method is called internally when the pointer moves over the chart, it just updates the crosshair position:

![alt](https://raw.githubusercontent.com/beto-rodriguez/LiveCharts2/dev//docs/samples/axes/crosshairs/result.gif)

You can call it if you require to update the crosshair.

**Reply by Eekkow** [reply]
Hello, thanks for your response. I have another question: when we update or redraw the x-axis, the labels change, but the label on the crosshair does not. Is this a bug, or is it the intended behavior? 

For instance, if the x-axis labels are initially 1, 2, 3, 4, 5, 6, 7, 8, 9, and the crosshair label shows '4,' when we change the data on the page, the x-axis labels are updated to 'Jan, Feb, March, April,' etc. However, the crosshair label still shows '4,' when we would expect it to update to match the new x-axis labels, such as 'April.' How can we ensure the crosshair label updates along with the x-axis labels?"

**Reply by HaganeNoRenkinjutsu** [reply]
> Hello, thanks for your response. I have another question: when we update or redraw the x-axis, the labels change, but the label on the crosshair does not. Is this a bug, or is it the intended behavior?
> 
> For instance, if the x-axis labels are initially 1, 2, 3, 4, 5, 6, 7, 8, 9, and the crosshair label shows '4,' when we change the data on the page, the x-axis labels are updated to 'Jan, Feb, March, April,' etc. However, the crosshair label still shows '4,' when we would expect it to update to match the new x-axis labels, such as 'April.' How can we ensure the crosshair label updates along with the x-axis labels?"

Hello Eakkow, given how old this post is you probably will never see my response, but I'll nonetheless respond anyway. Did you provide a Func to your x-axis labeler that translates the x-value coordinate? If you did not, then it's possible that the default Labeler doesn't know how to "translate" your x-axis data properly. Also, if you're wanting to use labels like Jan, Feb, March, etc, are you using a DateTimeAxis for your x-axis? To be honest, it's not possible for anyone to provide you support unless you provide the code that you've written. Information on labelers and DateTimeAxis is provided for on both the Livecharts2 website and the API documentation. The example samples are usually good enough to get you started on a solution to whatever problem you're facing.
